chirp/drivers/icr6.py

```diff
@@ -0,0 +1,794 @@
+            mem.offset = 9000 * ((_mem.offset_h * 256) + _mem.offset_l)
+        else:
+            print(f"Unknown freq multiplier: {_mem.freq_flags}")
+            mem.freq = 1234567890
+            mem.offset = 0
+
+        # mem.tuning_step = STEPS[_mem.tuning_step]
+        mem.duplex = DUPLEX_DIRS[_mem.duplex]
+
+        mem.ctone = TONES[_mem.ctone]
+        mem.tmode = TONE_MODES[_mem.tmode]
+
+        mem.mode = MODES[_mem.mode]
+
+        # memory scan skip
+        if _flag.skip == 0:
+            mem.skip = ""      # None
+        elif _flag.skip == 1:
+            mem.skip = "S"     # memscan skip (aka "Skip")
+        elif _flag.skip == 3:
+            mem.skip = "P"     # 3 = mem&vfo ("Pskip")
+
+        # Channel names are encoded in 6x 6-bit groups spanning 5 bytes.
+        # Mask only the needed bits and then lookup character for each group
+        # Mem format: 0000AAAA AABBBBBB CCCCCCDD DDDDEEEE EEFFFFFF
+        mem.name = CODED_CHRS[((_mem.name0 & 0x0F) << 2)
+                              | ((_mem.name1 & 0xC0) >> 6)] + \
+            CODED_CHRS[_mem.name1 & 0x3F] + \
+            CODED_CHRS[(_mem.name2 & 0xFC) >> 2] + \
+            CODED_CHRS[((_mem.name2 & 0x03) << 4)
+                       | ((_mem.name3 & 0xF0) >> 4)] + \
+            CODED_CHRS[((_mem.name3 & 0x0F) << 2)
+                       | ((_mem.name4 & 0xC0) >> 6)] + \
+            CODED_CHRS[(_mem.name4 & 0x3F)]
+        mem.name = mem.name.rstrip(" ").strip("^")
+
+        return mem
+
+    def get_settings(self):
+        """Translate the MEM_FORMAT structs into UI settings"""
+        # Based on the Icom IC-2730 driver
+        # Define mem struct write-back shortcuts
+        _sets = self._memobj.settings
+        _pses = self._memobj.pgmscanedge
+
+        basic = RadioSettingGroup("basic", "Basic Settings")
+        edges = RadioSettingGroup("edges", "Program Scan Edges")
+        common = RadioSettingGroup("common", "Common Settings")
+
+        group = RadioSettings(basic, edges, common)
+
+        # ----------------------
+        #   BASIC SETTINGS
+        # ----------------------
+
+        # ----------------------
+        #   PROGRAM SCAN EDGES
+        # ----------------------
+        for kx in range(0, 25):
+            stx = ""
+            for i in range(0, 6):
+                stx += chr(int(_pses[kx].name[i]))
+            stx = stx.rstrip()
+            rx = RadioSettingValueString(0, 6, stx)
+            rset = RadioSetting("pgmscanedge/%d.name" % kx,
+                                f"Program Scan {kx} Name", rx)
+            # rset.set_apply_callback(myset_psnam, _pses, kx, "name", 6)
+            edges.append(rset)
+
+            # Freq (Hz) is 1/3 the raw value (expect this allows N x 8.333kHz)
+            flow = 1.0 * (_pses[kx].start3 * 65536 * 256 +
+                          _pses[kx].start2 * 65536 +
+                          _pses[kx].start1 * 256 +
+                          _pses[kx].start0) / 3e6
```

Comment:
This almost certainly means you should be using a u32. Why not?

Reply:
Need to figure out the endian, kept on the "to do" list for now.